src/features/events/components/EventContactCard.tsx

```diff
@@ -0,0 +1,129 @@
+import CloseIcon from '@mui/icons-material/Close';
+import FaceOutlinedIcon from '@mui/icons-material/FaceOutlined';
+import FaceRetouchingOffOutlinedIcon from '@mui/icons-material/FaceRetouchingOffOutlined';
+import { Box, Button, Typography } from '@mui/material';
+import { FC, useContext } from 'react';
+
+import EventDataModel from 'features/events/models/EventDataModel';
+import messageIds from 'features/events/l10n/messageIds';
+import { useMessages } from 'core/i18n';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import ZUIAvatar from 'zui/ZUIAvatar';
+import ZUICard from 'zui/ZUICard';
+import { ZUIConfirmDialogContext } from 'zui/ZUIConfirmDialogProvider';
+import { MUIOnlyPersonSelect as ZUIPersonSelect } from 'zui/ZUIPersonSelect';
+
+interface EventContactCardProps {
+  data: ZetkinEvent;
+  model: EventDataModel;
+  orgId: number;
+}
+
+const EventContactCard: FC<EventContactCardProps> = ({
+  data,
+  model,
+  orgId,
+}) => {
+  const messages = useMessages(messageIds);
+  const { showConfirmDialog } = useContext(ZUIConfirmDialogContext);
+
+  const handleSelectedPerson = (personId: number) => {
+    model.setContact(personId);
+  };
+
+  return (
+    <Box
+      mb={2}
+      sx={{
+        '& div': {
+          flexDirection: 'column',
+          flexWrap: 'wrap',
+        },
+        display: 'block',
+      }}
+    >
+      <ZUICard
+        header={
+          <Box>
+            {model.getData().data?.contact?.id ? (
+              <Box>
+                <FaceOutlinedIcon sx={{ margin: 1, verticalAlign: 'middle' }} />
+                {messages.eventContactCard.header()}
+              </Box>
+            ) : (
+              <Box>
+                <FaceRetouchingOffOutlinedIcon
+                  sx={{ margin: 1, verticalAlign: 'middle' }}
+                />
+                {messages.eventContactCard.noContact()}
+              </Box>
+            )}
+          </Box>
+        }
+        status={
+          <Box>
+            {data.contact?.id ? (
+              <>
+                <Box
+                  m={1}
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  <ZUIAvatar orgId={orgId} personId={data.contact.id} />
+                </Box>
+                <Typography
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  {data.contact.name}
+                </Typography>
+              </>
+            ) : (
```

Comment:
I'm not a fan of these long ternary expressions because I find it difficult to see where the expressions start and end. Can it be refactored, e.g. as two components (defined in the same module), something like this:

```tsx
{!!data.contact ? (
  <ContactDetails contact={data.contact} />
) : (
  <ContactSelect onSelect={/*...*/} />
)}
```

Or using functions (that can be defined higher up in the component function):

```tsx
{!!data.contact ? (
  renderContactDetails(data.contact)
) : (
  renderContactSelect()
)}
```